Merge Mark's Changes for #143 into master (Retry)

JoelProminic opened the pull request:
Trying to merge @MarkProminic's changes from #143 again.  We tried to make this merge easier by handling the renames in a separate merge (#146), but that didn't simplify the GitHub action.

Also note that @MarkProminic is still making changes to this branch.   We need to get this wrapped up so that @piotrzarzycki21 can focus on the merge review.

piotrzarzycki21 commented:
Once @MarkProminic finish all I think it ends up with me reviewing manually his work. However PR should the way we should go, because when I will have some comments I think this will be the place where I will be putting it, regardless of the way of reviewing.

JoelProminic commented:
Since we seem to be getting close with @MarkProminic's branch, I regenerated the diff from the local client:

    Super.Human.Installer % git merge --no-commit --no-ff mark-testing-branch
    CONFLICT (rename/rename): Assets/provisioners/additional-prov/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_config/defaults/main.yml renamed to Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_config/defaults/main.yml in HEAD and to Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_config/defaults/main.yml in mark-testing-branch.
    CONFLICT (rename/rename): Assets/provisioners/additional-prov/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_genesis_applications/tasks/main.yml renamed to Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_genesis_applications/tasks/main.yml in HEAD and to Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_genesis_applications/tasks/main.yml in mark-testing-branch.
    CONFLICT (rename/rename): Assets/provisioners/additional-prov/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_jedi/tasks/main.yml renamed to Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_jedi/tasks/main.yml in HEAD and to Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_jedi/tasks/main.yml in mark-testing-branch.
    CONFLICT (rename/rename): Assets/provisioners/additional-prov/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2 renamed to Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2 in HEAD and to Assets/provisioners/hcl_domino_standalone_provisioner/0.1.20/scripts/ansible/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2 in mark-testing-branch.
    CONFLICT (rename/rename): Assets/provisioners/demo-tasks/0.1.20/scripts/ansible/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2 renamed to Assets/provisioners/hcl_domino_standalone_provisioner/0.1.20/scripts/ansible/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2 in HEAD and to Assets/provisioners/hcl_domino_standalone_provisioner/0.1.22/scripts/provisioners/ansible/roles/hcl_domino_traveler_htmo/templates/htmo-traveler-access.json.j2 in mark-testing-branch.
    CONFLICT (rename/rename): Assets/provisioners/demo-tasks/0.1.22/scripts/provisioners/ansible/roles/hcl_domino_traveler_htmo/templates/htmo-traveler-access.json.j2 renamed to Assets/provisioners/hcl_domino_standalone_provisioner/0.1.22/scripts/provisioners/ansible/roles/hcl_domino_traveler_htmo/templates/htmo-traveler-access.json.j2 in HEAD and to Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2 in mark-testing-branch.
    CONFLICT (rename/rename): Assets/provisioners/demo-tasks/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_config/defaults/main.yml renamed to Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_config/defaults/main.yml in HEAD and to Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_config/defaults/main.yml in mark-testing-branch.
    CONFLICT (rename/rename): Assets/provisioners/demo-tasks/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_genesis_applications/tasks/main.yml renamed to Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_genesis_applications/tasks/main.yml in HEAD and to Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_genesis_applications/tasks/main.yml in mark-testing-branch.
    CONFLICT (rename/rename): Assets/provisioners/demo-tasks/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_jedi/tasks/main.yml renamed to Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_jedi/tasks/main.yml in HEAD and to Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_jedi/tasks/main.yml in mark-testing-branch.
    CONFLICT (rename/rename): Assets/provisioners/demo-tasks/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2 renamed to Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2 in HEAD and to Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2 in mark-testing-branch.
    Auto-merging Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_config/defaults/main.yml
    CONFLICT (add/add): Merge conflict in Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_config/defaults/main.yml
    Auto-merging Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_genesis_applications/tasks/main.yml
    CONFLICT (add/add): Merge conflict in Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_genesis_applications/tasks/main.yml
    Auto-merging Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_jedi/tasks/main.yml
    CONFLICT (add/add): Merge conflict in Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_jedi/tasks/main.yml
    Auto-merging Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2
    CONFLICT (add/add): Merge conflict in Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2
    Auto-merging Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_config/defaults/main.yml
    CONFLICT (add/add): Merge conflict in Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_config/defaults/main.yml
    Auto-merging Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_genesis_applications/tasks/main.yml
    CONFLICT (add/add): Merge conflict in Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_genesis_applications/tasks/main.yml
    Auto-merging Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_jedi/tasks/main.yml
    CONFLICT (add/add): Merge conflict in Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_jedi/tasks/main.yml
    CONFLICT (rename/delete): Assets/provisioners/demo-tasks/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler/templates/gather_domino_traveler_internet_site_doc.json.j2 renamed to Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler/templates/gather_domino_traveler_internet_site_doc.json.j2 in HEAD, but deleted in mark-testing-branch.
    Auto-merging Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2
    CONFLICT (add/add): Merge conflict in Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2
    Auto-merging Source/SuperHumanInstaller.hx
    CONFLICT (content): Merge conflict in Source/SuperHumanInstaller.hx
    Auto-merging Source/superhuman/managers/ProvisionerManager.hx
    CONFLICT (content): Merge conflict in Source/superhuman/managers/ProvisionerManager.hx
    Auto-merging Source/superhuman/server/AdditionalServer.hx
    CONFLICT (content): Merge conflict in Source/superhuman/server/AdditionalServer.hx
    Auto-merging Source/superhuman/server/Server.hx
    CONFLICT (content): Merge conflict in Source/superhuman/server/Server.hx
    Auto-merging Source/superhuman/server/provisioners/ProvisionerType.hx
    CONFLICT (content): Merge conflict in Source/superhuman/server/provisioners/ProvisionerType.hx
    Auto-merging project.xml
    CONFLICT (content): Merge conflict in project.xml
    Automatic merge failed; fix conflicts and then commit the result.
    Super.Human.Installer % git status
    On branch master
    Your branch is up to date with 'origin/master'.

    You have unmerged paths.
      (fix conflicts and run "git commit")
      (use "git merge --abort" to abort the merge)

    Changes to be committed:
            modified:   .github/workflows/man-release.yml
            modified:   .github/workflows/sub-build-linux.yml
            modified:   .github/workflows/sub-build-macos.yml
            modified:   .github/workflows/sub-build-windows.yml
            modified:   .gitignore
            new file:   Assets/config/initial-registry.json
            new file:   Assets/domdownload.sh
            new file:   Assets/images/common/destroy_small.png
            new file:   Assets/images/common/power.png
            new file:   Assets/images/common/ssh.png
            modified:   Assets/images/common/start.png
            modified:   Assets/images/common/suspend.png
            modified:   Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/core/Hosts.rb
            new file:   Assets/provisioners/hcl_domino_additional_provisioner/provisioner.yml
            modified:   Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/core/Hosts.rb
            new file:   Assets/provisioners/hcl_domino_standalone_provisioner/provisioner.yml
            new file:   Assets/provisioners/provisioner-collection.yml.sample
            new file:   Assets/provisioners/provisioner.yml.sample
            modified:   Assets/text/en_US.json
            new file:   Genesis/Assets/images/common/add.png
            modified:   Genesis/Source/genesis/application/theme/GenesisApplicationTheme.hx
            modified:   Genesis/Source/prominic/sys/applications/bin/Shell.hx
            new file:   Genesis/Source/prominic/sys/applications/git/Git.hx
            modified:   Genesis/Source/prominic/sys/io/FileTools.hx
            modified:   Source/superhuman/browser/BrowserData.hx
            modified:   Source/superhuman/components/AdvancedConfigPage.hx
            modified:   Source/superhuman/components/ConfigPage.hx
            new file:   Source/superhuman/components/DownloadDialog.hx
            new file:   Source/superhuman/components/DynamicAdvancedConfigPage.hx
            new file:   Source/superhuman/components/DynamicConfigPage.hx
            new file:   Source/superhuman/components/HashManagerPage.hx
            new file:   Source/superhuman/components/ProvisionerImportPage.hx
            modified:   Source/superhuman/components/RolePage.hx
            new file:   Source/superhuman/components/SecretsPage.hx
            modified:   Source/superhuman/components/ServerList.hx
            modified:   Source/superhuman/components/ServerPage.hx
            modified:   Source/superhuman/components/SettingsPage.hx
            new file:   Source/superhuman/components/VerticalOptionsAlert.hx
            modified:   Source/superhuman/components/additionals/AdditionalServerPage.hx
            modified:   Source/superhuman/components/browsers/SetupBrowserPage.hx
            modified:   Source/superhuman/components/filesync/FileSyncSetting.hx
            modified:   Source/superhuman/components/filesync/FileSyncSetting.mxhx
            modified:   Source/superhuman/components/serviceType/ServiceTypePage.hx
            modified:   Source/superhuman/config/SuperHumanConfig.hx
            modified:   Source/superhuman/config/SuperHumanGlobals.hx
            modified:   Source/superhuman/config/SuperHumanHashes.hx
            modified:   Source/superhuman/config/SuperHumanPreferences.hx
            new file:   Source/superhuman/config/SuperHumanSecrets.hx
            new file:   Source/superhuman/downloaders/HCLDownloader.hx
            modified:   Source/superhuman/events/SuperHumanApplicationEvent.hx
            modified:   Source/superhuman/managers/ServerManager.hx
            new file:   Source/superhuman/server/cache/SuperHumanCachedFile.hx
            new file:   Source/superhuman/server/cache/SuperHumanFileCache.hx
            modified:   Source/superhuman/server/data/ServerData.hx
            modified:   Source/superhuman/server/data/ServiceTypeData.hx
            modified:   Source/superhuman/server/definitions/ProvisionerDefinition.hx
            modified:   Source/superhuman/server/hostsFileGenerator/AdditionalProvisionerHostsFileGenerator.hx
            renamed:    Source/superhuman/server/hostsFileGenerator/DemoTasksHostsFileGenerator.hx -> Source/superhuman/server/hostsFileGenerator/StandaloneProvisionerHostsFileGenerator.hx
            modified:   Source/superhuman/server/provisioners/AbstractProvisioner.hx
            modified:   Source/superhuman/server/provisioners/AdditionalProvisioner.hx
            new file:   Source/superhuman/server/provisioners/CustomProvisioner.hx
            renamed:    Source/superhuman/server/provisioners/DemoTasks.hx -> Source/superhuman/server/provisioners/StandaloneProvisioner.hx
            modified:   Source/superhuman/theme/SuperHumanInstallerTheme.hx
            modified:   Templates/installer/nsis-installer.nsi
            new file:   Templates/windows/app.manifest
            new file:   catalog.json
            new file:   token_request.json

    Unmerged paths:
      (use "git add/rm <file>..." as appropriate to mark resolution)
            both deleted:    Assets/provisioners/additional-prov/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_config/defaults/main.yml
            both deleted:    Assets/provisioners/additional-prov/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_genesis_applications/tasks/main.yml
            both deleted:    Assets/provisioners/additional-prov/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_jedi/tasks/main.yml
            both deleted:    Assets/provisioners/additional-prov/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2
            both deleted:    Assets/provisioners/demo-tasks/0.1.20/scripts/ansible/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2
            both deleted:    Assets/provisioners/demo-tasks/0.1.22/scripts/provisioners/ansible/roles/hcl_domino_traveler_htmo/templates/htmo-traveler-access.json.j2
            both deleted:    Assets/provisioners/demo-tasks/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_config/defaults/main.yml
            both deleted:    Assets/provisioners/demo-tasks/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_genesis_applications/tasks/main.yml
            both deleted:    Assets/provisioners/demo-tasks/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_jedi/tasks/main.yml
            both deleted:    Assets/provisioners/demo-tasks/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2
            both added:      Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_config/defaults/main.yml
            both added:      Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_genesis_applications/tasks/main.yml
            both added:      Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_jedi/tasks/main.yml
            both added:      Assets/provisioners/hcl_domino_additional_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2
            both added:      Assets/provisioners/hcl_domino_standalone_provisioner/0.1.20/scripts/ansible/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2
            both added:      Assets/provisioners/hcl_domino_standalone_provisioner/0.1.22/scripts/provisioners/ansible/roles/hcl_domino_traveler_htmo/templates/htmo-traveler-access.json.j2
            both added:      Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_config/defaults/main.yml
            both added:      Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_genesis_applications/tasks/main.yml
            both added:      Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_jedi/tasks/main.yml
            deleted by them: Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler/templates/gather_domino_traveler_internet_site_doc.json.j2
            both added:      Assets/provisioners/hcl_domino_standalone_provisioner/0.1.23/scripts/provisioners/ansible_collections/startcloud/hcl_roles/roles/domino_traveler_htmo/templates/htmo-traveler-access.json.j2
            both modified:   Source/SuperHumanInstaller.hx
            both modified:   Source/superhuman/managers/ProvisionerManager.hx
            both modified:   Source/superhuman/server/AdditionalServer.hx
            both modified:   Source/superhuman/server/Server.hx
            both modified:   Source/superhuman/server/provisioners/ProvisionerType.hx
            both modified:   project.xml

MarkProminic reviewed (commented):
I only had 6 files as merge conflicts, which I have resolved.

JoelProminic commented:
@MarkProminic reported that the changes are froxen for now, and this is ready for review.  The merge conflicts have been resolved.   Unfortunately, I'm still seeing 3000+ changed files for this merge, so our efforts to clear the rename issues were not successful.

From @MarkProminic's emails:

>Yep, I Don't need to make any more changes to make the app/features stable, and to get it deployed
>
>
>The only concern I really had was the file caching/worker threading (may get a beach ball when testing functiosn that involve moving files, give it some time)
>
>The import provisioner from Github, via HTTP is not fully implemented/working, you should currently only use Git as the default option for that.
>
>Other than those two minor things I tested all features/functions on Mac/Windows/Linux. 
>
>I have not however tested older provisioners like 0.1.22


In the current PR, I already handle the merge conflicts that arose

In VS code using the Pull request Extension, looks like it may be able to help parse the various files
